drivers/audio/dmic_nrfx_pdm.c

```diff
@@ -365,7 +365,6 @@ static int dmic_nrfx_pdm_configure(const struct device *dev,
 
 	channel->act_num_streams = 1;
 	channel->act_chan_map_hi = 0;
-	channel->act_chan_map_lo = def_map;
```

Comment:
Here is test with LEFT/Right channels swapped:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/audio/dmic_api/src/main.c#L158

Without this change, test is failing on this assert:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/audio/dmic_api/src/main.c#L94

In current/previous implementation, actual channel map is always `def_map`.
Even if the user requested values equal to the `alt_map`.